zhmcclient/_partition.py

```diff
@@ -1550,6 +1550,519 @@ def list_attached_storage_groups(self, full_properties=False):
+                    raise PartitionLinkError(result['operation-results'])
+            return None
+
+        return result  # Job
+
+    @logged_api_call
+    def list_attached_partition_links(
+            self, type=None, name=None, state=None, additional_properties=None):
+        # pylint: disable=redefined-builtin
+        """
+        Return the partition links attached to this partition, optionally
+        filtered by partition link type, name and status.
+
+        This method performs the "List Partition Links" HMC operation.
+
+        HMC/SE version requirements:
+
+        * SE version >= 2.16.0
+        * for technology-specific support, see the API features described in
+          :ref:`Partition Links`
+
+        Authorization requirements:
+
+        * Object-access permission to this partition.
+        * Object-access permission to all partition links returned (only those
+          are returned).
+
+        Parameters:
+
+          type (:term:`string`): Filter string to limit returned partition
+            links to those with the specified type. Valid values are: "smc-d",
+            "hipersockets", "ctc". If `None`, no filtering for the partition
+            link type takes place.
+
+          name (:term:`string`): Filter pattern (regular expression)
+            to limit returned partition links to those that have a matching
+            name. If `None`, no filtering for the partition link name takes
+            place.
+
+          state (:term:`string`): Filter string to limit returned partition
+            links to those with the specified state. Valid values are:
+            "complete", "incomplete", "updating". If `None`, no filtering for
+            the partition link state takes place.
+
+          additional_properties (list of string): Additional properties to
+            be included in the returned :class:`~zhmcclient.PartitionLink`
+            objects, in addition to the minimum list of ('cpc-uri', 'state',
+            'name', 'type', 'object-uri'). The properties must be specified
+            with hyphens in their names (not underscores).
+
+        Returns:
+
+          List of :class:`~zhmcclient.PartitionLink` objects representing the
+          partition links attached to this partition.
+
+        Raises:
+
+          :exc:`~zhmcclient.HTTPError`
+          :exc:`~zhmcclient.ParseError`
+          :exc:`~zhmcclient.AuthError`
+          :exc:`~zhmcclient.ConnectionError`
+        """
+        cpc = self.manager.cpc
+        console = cpc.manager.console
+
+        filter_args = {
+            'cpc-uri': cpc.uri,
+        }
+        if type:
+            filter_args['type'] = type
+        if name:
+            filter_args['name'] = name
+        if state:
+            filter_args['state'] = state
+
+        pl_list = console.partition_links.list(
+            filter_args=filter_args,
+            additional_properties=additional_properties)
```

Comment:
Can we always add 'paths' and 'bus-connections' to additional_properties for improved performance as they are used for the filtering below?

Reply:
DONE